src/pages/settings/Security/AddDelegate/ConfirmDelegatePage.tsx

```diff
@@ -73,20 +73,18 @@ function ConfirmDelegatePage({route}: ConfirmDelegatePageProps) {
                 onPress={() => Navigation.navigate(ROUTES.SETTINGS_DELEGATE_ROLE.getRoute(login, role))}
                 shouldShowRightIcon
             />
-
-            {isValidateCodeActionModalVisible && (
-                <DelegateMagicCodeModal
-                    login={login}
-                    role={role}
-                    onClose={() => {
-                        if (!showValidateActionModal) {
-                            return;
-                        }
-
-                        Navigation.navigate(ROUTES.SETTINGS_SECURITY);
-                    }}
-                />
-            )}
+            <DelegateMagicCodeModal
+                login={login}
+                role={role}
+                onClose={() => {
+                    setIsValidateCodeActionModalVisible(false);
+                    if (!showValidateActionModal) {
+                        return;
+                    }
+                    Navigation.navigate(ROUTES.SETTINGS_SECURITY);
```

Comment:
@cretadn22 It seems this case was added to handle navigation depending on the source page. If we arrive from the `SecuritySettingsPage` and close the `DelegateMagicCodeModal`, we need to navigate back to the `SecuritySettingsPage` on close. However, if we come from the Add Copilot flow, we should simply hide the modal without further navigation.

Reply:
After reviewing the `git blame` and testing the flow, I believe a bit of refactoring is needed here. For example, we can remove the entire usage of `showValidateActionModal` and use `DelegateMagicCodeModal` directly on the `SecuritySettingsPage`

Reply:
@getusha Could you help us understand why we navigate to the `DelegateConfirmPage` from the `SecuritySettingsPage`? Wouldn’t it be easier to handle if we used the `DelegateMagicCodeModal` directly on the `SecuritySettingsPage` instead? Or am I missing some underlying logic here?

https://github.com/Expensify/App/blob/6cd5c645c4e3ad6e0fff934457d1c42a1b5b209e/src/pages/settings/Security/SecuritySettingsPage.tsx#L157

Reply:
@cretadn22, I've pinged Getabalew on [Slack](https://expensify.slack.com/archives/C02NK2DQWUX/p1730291343417389) to see why the above was added and what it does. Let's see if they can help us understand.

cc: @carlosmiceli

Reply:
Strange, I was testing, and I was able to use the `DelegateMagicCodeModal` on the `SecuritySettingsPage`. It opened, but I faced one issue when I directly used the `DelegateMagicCodeModal` on the `SecuritySettingsPage`: when we pressed the back icon on the `DelegateMagicCodeModal`, there was no back navigation; the modal closed without animation

Reply:
@cretadn22 Please try to open the DelegateMagicCodeModal on the SecuritySettingsPage on your side as well. Let us know whether you are able to do so.

Reply:
> I would like to propose this solution. What is your opinion on it?

That will break the current navigation flow, If we come to ConfirmDelegatePage from SecuritySettingsPage

Reply:
> I am unclear about your meaning. Could you please elaborate on https://github.com/Expensify/App/pull/51350#discussion_r1822547707?

In the code, try to check the `showValidateActionModal` logic; you may get an idea of what I am trying to say.

Reply:
@cretadn22 please upload a video for this flow, after applying your changes:

1. Add copilot -> Select Access level -> Press Add copilot
2. Enter invalid magic code
3. Close RHP
4. Press on the user that we can see on Security page under Delegated access
5. navigated to magic code page
6. Press back from header

Reply:
this is existing flow, please upload a video of same flow from your PR so that we can compare side by side
https://github.com/user-attachments/assets/4789c492-062f-44fe-85e0-29363310e1bb

Reply:
> If so, I believe the current code is both safer and simpler.

Can we proceed by adding `setIsValidateCodeActionModalVisible` in the `onClose` function?

Reply:
Oh, sorry, I missed your commit. Have you tested all the platforms again after this change?

Reply:
One question: can we add `setIsValidateCodeActionModalVisible(false);` inside the `if (!showValidateActionModal) {` block? Since, if `showValidateActionModal` is true, we’re already calling the navigation.navigate, so when we return to this page, `isValidateCodeActionModalVisible` will be set to true.